File: patch_brightness.py
import os

filepath = 'src/components/viz/BrightnessMeter.test.jsx'
with open(filepath, 'r') as f:
    content = f.read()

old_mock = """vi.mock('lucide-react', () => {
    const React = require('react');
    const createIcon = (name) => (props) => React.createElement('div', { ...props, 'data-testid': name });

    return {
        Sun: createIcon('Sun'),
        Moon: createIcon('Moon'),
        Info: createIcon('Info'),
        Smile: createIcon('Smile')
    };
});"""

new_mock = """vi.mock('lucide-react', async () => {
    const React = await import('react');
    const createIcon = (name) => {
        const Icon = (props) => React.createElement('div', { ...props, 'data-testid': name });
        Icon.displayName = name;
        return Icon;
    };

    return {
        Sun: createIcon('Sun'),
        Moon: createIcon('Moon'),
        Info: createIcon('Info'),
        Smile: createIcon('Smile')
    };
});"""

content = content.replace(old_mock, new_mock)

with open(filepath, 'w') as f:
    f.write(content)

File: patch_brightness.js
import fs from 'fs';

let content = fs.readFileSync('src/components/viz/BrightnessMeter.test.jsx', 'utf8');
content = content.replace(
    "    const React = require('react');\n    const createIcon = (name) => (props) => React.createElement('div', { ...props, 'data-testid': name });\n\n    return {\n        Sun: createIcon('Sun'),\n        Moon: createIcon('Moon'),\n        Info: createIcon('Info'),\n        Smile: createIcon('Smile')\n    };\n});",
    `    const createIcon = (name) => {
        const MockComponent = function MockComponent(props) {
            return <div data-testid={name} {...props} />;
        };
        MockComponent.displayName = name;
        return MockComponent;
    };

    return {
        Sun: createIcon('Sun'),
        Moon: createIcon('Moon'),
        Info: createIcon('Info'),
        Smile: createIcon('Smile')
    };
});`
);
fs.writeFileSync('src/components/viz/BrightnessMeter.test.jsx', content);
